internal/runtime_reflect/map.go

```diff
@@ -0,0 +1,87 @@
+package runtime_reflect
```

Comment:
Can you give a tl;dr in the package description about why we're dropping down this low? And maybe a warning indicating that it relies on implementation details (e.g. the layout of the `hiter` struct) which may change between go versions.